src/eth/provider.ts

```diff
@@ -55,3 +56,16 @@ export async function disconnect(): Promise<void> {
     return
   }
 }
+
+export async function switchToChainId(chainId: ChainId) {
+  try {
+    const provider = await connection.getProvider()
+
+    await switchProviderChainId(provider, chainId)
+    window.location.reload()
```

Comment:
I was wondering, is this reload necessary? I mean, if we're changing the current provider, is it possible to just do ir without reloading the entire page?